lib/utils.js

```diff
@@ -1,134 +1,46 @@
- * @return {boolean} An indicator of whether the data is valid
- */
-exports.verifyDataFormat = function (data) {
-  if (!data || data === {}) {
-    let isValid = false;
-    let failures = {'no data provided': 1};
-    return { isValid, failures };
-  }
-  let v = validate(data, dataSchema);
-  let isValid = v.errors.length === 0;
-  let failures = [];
-  v.errors.forEach((e) => {
-    failures.push(e.stack);
-  });
-
-  return { isValid, failures };
-};
-
-exports.verifyStream = function (stream) {
-  if (typeof stream !== 'object' && stream instanceof DataStream) {
-    let isValid = false;
-    let failures = {'no data provided': 1};
-    return { isValid, failures };
-  }
-  let isValid = !!stream.name;
-  let failures = [];
-
-  return { isValid, failures };
-};
-
-/**
- * Prune measurements that don't meet our requirements.
- * @param {array} measurements The measurements array to prune measurements from
- * @return { pruned, failures } An array pruned of invalid measurement objects,
- * may be empty and a failures object of aggregated reasons for data failures
- */
-exports.pruneMeasurements = function (measurements) {
-  let failures = [];
-  let pruned = filter(measurements, function (m) {
-    let v = validate(m, measurementSchema);
-    v.errors.forEach((e) => {
-      failures.push(e.stack);
-    });
-
-    // Return false if anything failed
-    if (v.errors.length >= 1) {
-      return false;
-    }
-
-    return true;
-  });
-
-  // Aggregate failures
-  failures = countBy(failures);
-
-  return { pruned, failures };
-};
-
-/**
- * Removes unwanted measurement types by parameters
- * @param {array} measurements An array of measurements to remove unwanted
- * paramters from.
- * @return An array of measurements of desired paramters
- */
-exports.removeUnwantedParameters = function (measurements) {
-  let validParams = ['pm25', 'pm10', 'no2', 'so2', 'o3', 'co', 'bc'];
-  let desired = filter(measurements, (m) => {
-    return validParams.indexOf(m.parameter) !== -1;
-  });
-
-  return desired;
-};
+import { FetchError, STREAM_END } from './errors';
+import log from './logger';
 
 /**
  * Convert units into system-preferred units.
  * @summary The preferred unit for mass concentration is 'µg/m3', for volumetric
  * concentrations this is 'ppm'.
- * @param {array} measurements An array of measurements to potentially convert units of
- * @return An array of measurements converted to system-preferred units
+ * @param {Array} measurements An array of measurements to potentially convert units of
+ * @return {Array} An array of measurements converted to system-preferred units
  */
-exports.convertUnits = function (measurements) {
-  if (!measurements) {
-    return;
-  }
-
-  for (var i = 0; i < measurements.length; i++) {
-    var m = measurements[i];
-
-    // Skip if we don't have a valid measurement
-    if (!m) {
-      continue;
-    }
+export function convertUnits (array) {
+  array.filter(x => x).forEach(unifyMeasurementUnits);
```

Comment:
`x => x` in `filter` is simply to filter out falsy values, especially nulls or undefined. Those units cannot be unified, since there's nothing there to unify. ;)

BTW, I've taken the liberty of removing this method in a next commit (I left a stub, so that legacy adapters won't break). Instead it's being run by the fetch process, along with a couple other checks (like removing unwanted measurements).